dashboard/app/models/user.rb

```diff
@@ -605,10 +605,43 @@ def user_level_locked?(script_level, level)
     user_level.nil? || user_level.locked?(script_level.stage)
   end
 
+  # Returns the level in this script that the user most recently made progress on
+  def last_progressed_level(script)
+    user_levels.
+      where(script_id: script.id).
+      sort_by(&:updated_at).
+      last
+  end
+
+  # Returns the next script_level for the next progression level in the given
+  # script that hasn't yet been passed, starting its search at the last level we submitted
```

Comment:
I believe that historically we (a) only used async progress writing during HoC and (b) we chose not to use it during last HoC

In the case that we were to turn on async progress writing, this feature would not work as expected, but I don't think that is all that bad of a bad.